File: pascal_triangle/pascalTriangle.js
/**
 * Pascal triangle
 */

function pascalTriangle(n) {
  const triangle = [];
  for( let i = 0; i < n; i++) {
    const rowList = [];
    for(let j = 0; j <= i; j++){
      if(j === 0 || j === i){
        rowList.push(1);
      } else {
        const col = triangle[i - 1][j - 1] + triangle[i - 1][j];
        rowList.push(col);
      }
    }
    triangle.push(rowList);
  }

  for (let i = 0; i < n; i++) {
    let spaces = ' '.repeat(n - i - 1);
    process.stdout.write(spaces);
    for (let j = 0; j < i + 1; j++) {
        process.stdout.write(triangle[i][j].toString().padEnd(4, ' '));
    }
    process.stdout.write('\n');
  }
}

let cas = pascalTriangle(4)
cas = pascalTriangle(6)
cas = pascalTriangle(10)

File: pascal_triangle/pascal_triangle.py
"""
Pascal Triangle
"""

def pascal_triangle(n):
  list = []

  for i in range(n):
    row_list = []
    for j in range(i + 1):
      if j == 0 or j == i:
        row_list.append(1)
      else:
        col = list[i - 1][j - 1] + list[i - 1][j]
        row_list.append(col)
    list.append(row_list)

  for i in range(n):
    for j in range(n-i-1):
      print(format(' ', '<2'), end='')
    for j in range(i+1):
      print(format(list[i][j], '<3'), end=' ')
    print()



case1 = pascal_triangle(4)
case1 = pascal_triangle(6)
case1 = pascal_triangle(10)